src/Crc.cpp

```diff
@@ -32,17 +31,12 @@ uint16_t Crc::crc16(uint8_t* bytes, uint32_t length) {
     return crc & 0xffff;
 }
 
-uint32_t Crc::crc32(const Data& data)
-{
-    boost::crc_32_type result;
-    result.process_bytes((const void*)data.data(), data.size());
-    return (uint32_t)result.checksum();
-}
-
-uint32_t Crc::crc32C(const Data& data)
-{
-    using crc_32c_type = boost::crc_optimal<32, 0x1EDC6F41, 0xFFFFFFFF, 0xFFFFFFFF, true, true>;
-    crc_32c_type result;
-    result.process_bytes((const void*)data.data(), data.size());
-    return (uint32_t)result.checksum();
+// Algorithm inspired by this old-style C implementation:
+// https://web.mit.edu/freebsd/head/sys/libkern/crc32.c (Public Domain code)
+uint32_t Crc::crc32(const Data& data) {
+    uint32_t c = std::numeric_limits<uint32_t>::max();
+    for (const auto byte : data) {
```

Comment:
Here, the bytes are copied at each iteration - to let the compiler choose the best option for u I recommend:

```cpp
for (auto&& byte: data)
```

Reply:
My thinking was that since this is a primitive data type uint8_t, copying by reference is unnecessary and potentially has a larger overhead.

Reply:
Yes but it's more flexible: https://godbolt.org/z/3WsdvjrcP the assembly will be the same, the compiler will choose the best option (here a copy) and if in the future Data change the underlying type, the compiler will pick up the best solution automatically.

Reply:
Okay